version: "3.9"

services:
  mysql:
    container_name: mysql
    image: mysql:8.0
    platform: linux/amd64
    environment:
      MYSQL_ROOT_PASSWORD: "P@55W0rD"
    ports:
      - "3366:3306"
    healthcheck:
      test: [ "CMD", "mysqladmin" ,"ping", "-h", "localhost" ]
      interval: 5s
      timeout: 10s
      retries: 10
    volumes:
      - ./docker/mysql/data:/var/lib/mysql
      - ./docker/mysql/script:/opt/scripts
      - ./docker/mysql/script/setup.sql:/docker-entrypoint-initdb.d/1.sql
    command: --default-authentication-plugin=mysql_native_password
    networks:
      - purchase-network

  localstack:
    container_name: localstack
    image: fiapsoat2grupo13/localstack-resources:latest
    platform: linux/amd64
    ports:
      - "127.0.0.1:4566:4566"            # LocalStack Gateway
      - "127.0.0.1:4510-4559:4510-4559"  # external services port range
    environment:
      # LocalStack configuration: https://docs.localstack.cloud/references/configuration/
      - DEBUG=${DEBUG:-0}
    volumes:
      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
    networks:
      - purchase-network

networks:
  purchase-network:
    driver: bridge

